File: sic-api/Entities/Su/SuBusinessRole.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace sic_api.Entities.Su;

[Index(nameof(BusinessId), nameof(RoleCode), IsUnique = true)]
[Table("su_business_role")]
public class SuBusinessRole : BaseEntity
{
    [Required]
    [ForeignKey(nameof(Business))]
    [Column("business_id")]
    public Guid BusinessId { get; set; }
    public SuBusiness Business { get; set; } = default!;

    [ForeignKey(nameof(ParentRole))]
    [Column("parent_role_id")]
    public Guid? ParentRoleId { get; set; }
    public SuBusinessRole? ParentRole { get; set; }

    [Required]
    [MaxLength(50)]
    [Column("role_code")]
    public string RoleCode { get; set; } = default!;

    [Required]
    [MaxLength(255)]
    [Column("role_name_en")]
    public string RoleNameEn { get; set; } = default!;

    [Required]
    [MaxLength(255)]
    [Column("role_name_local")]
    public string RoleNameLocal { get; set; } = default!;

    [MaxLength(50)]
    [Column("role_level")]
    public string? RoleLevel { get; set; }

    [Column("sort_order")]
    public int? SortOrder { get; set; }

    [Column("is_active")]
    public bool IsActive { get; set; } = true;

    public ICollection<SuBusinessRole> ChildRoles { get; set; } = new List<SuBusinessRole>();
    public ICollection<SuBusinessRoleProgram> RolePrograms { get; set; } = new List<SuBusinessRoleProgram>();
    public ICollection<SuUserBusinessRole> UserBusinessRoles { get; set; } = new List<SuUserBusinessRole>();
    public ICollection<SuBusinessInvite> Invites { get; set; } = new List<SuBusinessInvite>();
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/su/SuBusinessRole.java
package com.softinter.sicapi.entity.su;

import com.softinter.sicapi.entity.base.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "su_business_role",
       indexes = {
           @Index(name = "idx_business_role_code", columnList = "business_id, role_code", unique = true)
       })
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class SuBusinessRole extends BaseEntity {

    // Foreign key fields (explicit to match C# property names)
    @Column(name = "business_id", insertable = false, updatable = false)
    private UUID businessId;

    @Column(name = "parent_role_id", insertable = false, updatable = false)
    private UUID parentRoleId;

    // Navigation properties
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "business_id", nullable = false)
    private SuBusiness business;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_role_id")
    private SuBusinessRole parentRole;

    @Column(name = "role_code", nullable = false, length = 50)
    private String roleCode;

    @Column(name = "role_name_en", nullable = false, length = 255)
    private String roleNameEn;

    @Column(name = "role_name_local", nullable = false, length = 255)
    private String roleNameLocal;

    @Column(name = "role_level", length = 50)
    private String roleLevel;

    @Column(name = "sort_order")
    private Integer sortOrder;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = true;

    @Column(name = "color", length = 20)
    private String color;

    // Collections
    @OneToMany(mappedBy = "parentRole", fetch = FetchType.LAZY)
    private List<SuBusinessRole> childRoles = new ArrayList<>();

    @OneToMany(mappedBy = "businessRole", fetch = FetchType.LAZY)
    private List<SuBusinessRoleProgram> rolePrograms = new ArrayList<>();

    @OneToMany(mappedBy = "businessRole", fetch = FetchType.LAZY)
    private List<SuUserBusinessRole> userBusinessRoles = new ArrayList<>();

    @OneToMany(mappedBy = "suBusinessRole", fetch = FetchType.LAZY)
    private List<SuBusinessInvite> invites = new ArrayList<>();
}
